docs/src/simulation.md

```diff
@@ -0,0 +1,183 @@
+Simulating an Experiment from Scratch
+======================================
+
+Here is a worked example of simulating a partially crossed design from scratch.
+
+First, some setup:
+
+```@example Main
+using DataFrames
+using LinearAlgebra
+using MixedModels, MixedModelsSim
+using Random
+```
+
+## Assemble the Design
+
+We're going to do a 2 x 2 x 2 design.
+For concreteness, let's think of this as a linguistic design:
+
+- **age** `old` vs. `young`, _between subjects_
+- **frequency** `high` vs. `low`, _between items_
+- **context** `matched` vs. `unmatched`, _within both_.
+
+Further, let's assume we want 40 subjects and 80 items.
+We can specify this design as follows:
+
+```@example Main
+n_subj = 40
+n_item = 80
+subj_btwn = Dict(:age => ["old", "young"])
+item_btwn = Dict(:frequency => ["high", "low"])
+both_win = Dict(:context => ["matched", "unmatched"])
+```
+
+and then generate it:
+
+```@example Main
+rng = MersenneTwister(42)  # specify our random number generator for reproducibility
+design = simdat_crossed(rng, n_subj, n_item;
+                        subj_btwn = subj_btwn,
+                        item_btwn = item_btwn,
+                        both_win = both_win)
+```
+
+Note that `simdat_crossed` returns a row table, which `MixedModels.jl` can process directly.
+For nicely displaying it, we can also convert it to a dataframe and change the factors to use pooled arrays to save a bit of memory.
+
+```@example Main
+df = pooled!(DataFrame(design))
+first(df, 5)
```

Comment:
At this point it seems that `df.subj` has 80 levels and `df.item` has 160.
```julia
julia> describe(df)
6×7 DataFrame
 Row │ variable   mean        min       median      max        nmissing  eltype   
     │ Symbol     Union…      Any       Union…      Any        Int64     DataType 
─────┼────────────────────────────────────────────────────────────────────────────
   1 │ subj                   S01                   S80               0  String
   2 │ age                    old                   young             0  String
   3 │ item                   I001                  I160              0  String
   4 │ frequency              high                  low               0  String
   5 │ context                matched               unmatched         0  String
   6 │ dv         0.00239127  -4.11127  0.00685659  3.93649           0  Float64
```
I can look into why, if you wish.

Reply:
I think this is because `simdat_crossed` aims for `n` items/subjects per condition for between designs. So for 40 subjects and 80 items, each with a 2-level between factor, we get 80 and 160. Maybe we should just document that logic?

Reply:
It seems kind of awkward to me if the first argument is `nsubj` and it doesn't end up being the number of subjects.  If the number of levels of the between factor divides the number of subjects then perhaps we should just divide the number of subjects and number of items by that for the construction.

Reply:
I think I've fixed this now

Reply:
Thanks